edge-functions/feature-flag-optimizely/pages/_middleware.tsx

```diff
@@ -1,22 +1,19 @@
 //@ts-ignore
 import { createInstance } from '@optimizely/optimizely-sdk/dist/optimizely.lite.min.js'
+import { v4 } from 'uuid'
 import { NextRequest, NextFetchEvent, NextResponse } from "next/server"
 import optimizelyDatafile from '../lib/optimizely/datafile.json'
 
 const VERCEL_EDGE_CLIENT_ENGINE = 'javascript-sdk/vercel-edge'
 const COOKIE_NAME = 'optimizely_visitor_id'
 
-function generateRandomUserId(): string {
-  return Math.floor(Math.random() * 899999 + 100000) as unknown as string
-}
-
 export async function middleware(req: NextRequest, ev: NextFetchEvent) {
   if (!req.page.name) {
     return NextResponse.next()
   }
 
   // Fetch user Id from the cookie if available so a returning user from same browser session always sees the same variation.
-  const userId = req.cookies[COOKIE_NAME] || generateRandomUserId()
+  const userId = req.cookies[COOKIE_NAME] || v4()
```

Comment:
Can we switch this to use [crypto.randomUUID()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)?

Reply:
Done!